src/api/app/models/event/base.rb

```diff
@@ -273,6 +273,12 @@ def _roles(role, project, package = nil)
       obj_roles(p, role)
     end
 
+    def send_to_bus
+      RabbitmqBus.publish(self.class.message_bus_queue, read_attribute(:payload))
+    rescue Bunny::Exception => e
+      logger.error "Publishing to RabbitMQ failed: #{e.message}"
```

Comment:
there are 2 cases: when you don't even have a bus configured - then publish will return peacefully and we can continue.

But if it fails to publish to a configured bus, our whole tooling concept collapses - the idea is that tools can rely on the bus and can stop polling. But if events can be dropped so easily, they will keep polling.

In my PR I mainly put them in the queue because I replaced the send_to_backend method 1:1, but we don't have to. A possible model can also be to send asap and only if it fails, put it in a delayed queue.

The catch with the asap model is that the bus becomes your performance bottleneck - but rabbitmq is core to so many architectures, that I don't believe it's a performance problem.

Reply:
I see. Thanks for the explanation:-)

> coolo 21 minutes ago 
> A possible model can also be to send asap and only if it fails, put it in a delayed queue.

That sounds like a good solution to me. @openSUSE/open-build-service What do you say?

Reply:
That was what we had in mind when started this PR. Wasn't included here but was one of the things to do. So, yes from our side, we want to deliver ASAP and in case of failure put that in a delayed job.

Reply:
> A possible model can also be to send asap and only if it fails, put it in a delayed queue.

@coolo How likely is it that the rabbitmq server will fail? Is it not fair to say to OBS admins that if their rabbitmq server goes down then they won't get events sent to it? 

The reason why I'm asking this is because I would prefer not to use delayed jobs to handle events which dont get sent because rabbitmq is down. We've had so many problems with the delayed jobs recently I would prefer to avoid using them unless absolutely necessary.

Reply:
> A possible model can also be to send asap and only if it fails, put it in a delayed queue.

There are ways to deal with reliable delivery of messages in MQ. However you have to decide how big the hammer needs to be....

If you want the OBS bus to be 100% reliable you have to use [publisher](http://rubybunny.info/articles/extensions.html#publisher_confirms_publisher_acknowledgements) and [message](http://rubybunny.info/articles/queues.html#message_acknowledgements) acknowledgements.

I would like to avoid to make our code overly complicated for making publisher<->broker 100% reliable without people dealing with the rest that involves making the broker setup [distributed](http://www.rabbitmq.com/distributed.html), [HA](http://www.rabbitmq.com/ha.html) and by introducing other [MQ reliability](http://www.rabbitmq.com/reliability.html) concepts...

IMHO either we are getting all into MQ reliable delivery or we don't...

Reply:
If hermes is down, we keep sending too.

Reply:
The difference is that sending shit to hermes is not MQ and does not use a broker in between right? :-)